pkg/injector/webhook.go

```diff
@@ -120,77 +120,75 @@ func (wh *webhook) run(stop <-chan struct{}) {
+func healthHandler(w http.ResponseWriter, _ *http.Request) {
 	w.WriteHeader(http.StatusOK)
-	_, err := w.Write([]byte("Health OK"))
-	if err != nil {
-		log.Error().Err(err).Msgf("Error writing bytes")
+	if _, err := w.Write([]byte("Health OK")); err != nil {
+		log.Error().Err(err).Msgf("Error writing bytes for mutating webhook health check handler")
 	}
 }
 
 func (wh *webhook) mutateHandler(w http.ResponseWriter, req *http.Request) {
-	log.Info().Msgf("Request received: Method=%v, URL=%v", req.Method, req.URL)
+	log.Trace().Msgf("Received mutating webhook request: Method=%v, URL=%v", req.Method, req.URL)
 
-	if contentType := req.Header.Get("Content-Type"); contentType != "application/json" {
-		errmsg := fmt.Sprintf("Invalid Content-Type: %q", contentType)
-		http.Error(w, errmsg, http.StatusUnsupportedMediaType)
-		log.Error().Msgf("Request error: error=%s, code=%v", errmsg, http.StatusUnsupportedMediaType)
+	if contentType := req.Header.Get("Content-Type"); contentType != contentTypeJSON {
+		err := errors.Errorf("Invalid content type: %s; Expected %s", contentType, contentTypeJSON)
+		http.Error(w, err.Error(), http.StatusUnsupportedMediaType)
+		log.Error().Err(err).Msgf("Responded to admission request with HTTP %v", http.StatusUnsupportedMediaType)
 		return
 	}
 
-	var body []byte
+	var admissionRequestBody []byte
 	if req.Body != nil {
 		var err error
-		if body, err = ioutil.ReadAll(req.Body); err != nil {
-			errmsg := fmt.Sprintf("Error reading request body: %s", err)
-			http.Error(w, errmsg, http.StatusInternalServerError)
-			log.Error().Msgf("Request error: error=%s, code=%v", errmsg, http.StatusInternalServerError)
+		if admissionRequestBody, err = ioutil.ReadAll(req.Body); err != nil {
+			err := errors.Errorf("Error reading request admissionRequestBody: %s", err)
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			log.Error().Err(err).Msgf("Responded to admission request with HTTP %v", http.StatusInternalServerError)
 			return
 		}
 	}
 
-	if len(body) == 0 {
-		errmsg := "Empty request body"
-		http.Error(w, errmsg, http.StatusBadRequest)
-		log.Error().Msgf("Request error: error=%s, code=%v", errmsg, http.StatusBadRequest)
+	if len(admissionRequestBody) == 0 {
+		err := errors.New("Empty request admissionRequestBody")
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		log.Error().Err(err).Msgf("Responded to admission request with HTTP %v", http.StatusBadRequest)
 		return
 	}
 
 	var admissionReq v1beta1.AdmissionReview
 	var admissionResp v1beta1.AdmissionReview
-	if _, _, err := deserializer.Decode(body, nil, &admissionReq); err != nil {
-		log.Error().Err(err).Msg("Error decoding admission request")
-		admissionResp.Response = toAdmissionError(err)
+	if _, _, err := deserializer.Decode(admissionRequestBody, nil, &admissionReq); err != nil {
+		log.Error().Err(err).Msgf("Error decoding admission request body: %s", string(admissionRequestBody))
```

Comment:
The `admissionRequestBody` could contain sensitive data. Maybe consider logging the error but not the whole request body?

Reply:
Corrected